mmpose/core/evaluation/top_down_eval.py

```diff
@@ -91,6 +91,43 @@ def _get_max_preds(heatmaps):
     return preds, maxvals
 
 
+def _get_max_preds_3d(heatmaps):
+    """Get keypoint predictions from 3D score maps.
+
+    Note:
+        batch size: N
+        num keypoints: K
+        heatmap depth size: D
+        heatmap height: H
+        heatmap width: W
+
+    Args:
+        heatmaps (np.ndarray[N, K, D, H, W]): model predicted heatmaps.
+
+    Returns:
+        tuple: A tuple containing aggregated results.
+        - preds (np.ndarray[N, K, 3]): Predicted keypoint location.
+        - maxvals (np.ndarray[N, K, 1]): Scores (confidence) of the keypoints.
+    """
+    assert isinstance(heatmaps, np.ndarray), \
+        ('heatmaps should be numpy.ndarray')
+    assert heatmaps.ndim == 5, 'heatmaps should be 5-ndim'
+
+    N, K, D, H, W = heatmaps.shape
+    heatmaps_reshaped = heatmaps.reshape((N, K, -1))
+    idx = np.argmax(heatmaps_reshaped, 2).reshape((N, K, 1))
+    maxvals = np.amax(heatmaps_reshaped, 2).reshape((N, K, 1))
+
+    preds = np.zeros((N, K, 3), dtype=np.float32)
+    _idx = idx[..., 0]
+    preds[:, :, 2], _idx = _idx // (H * W), _idx % (H * W)
+    preds[:, :, 1], _idx = _idx // W, _idx % W
+    preds[:, :, 0] = _idx
```

Comment:
Do you think the following form would be more clear to users? Just a suggestion.

preds[...,2] = _idx // (H*W)
preds[...,1] = (_idx // W) % H
preds[...,0] = _idx % W

Reply:
OK, modified.